src/main/java/com/powsybl/openloadflow/ac/equations/AcEquationSystem.java

```diff
@@ -318,104 +335,154 @@ private static void createBranchActivePowerTargetEquation(LfBranch branch, Discr
-                                .addTerm(equationSystem.getVariable(controllerBranch.getNum(), AcVariableType.BRANCH_RHO1).createTerm())
-                                .setActive(false);
+                                .addTerm(equationSystem.getVariable(controllerBranch.getNum(), AcVariableType.BRANCH_RHO1).createTerm());
                     }
+
+                    updateTransformerVoltageControlEquations(voltageControl, equationSystem);
                 });
+    }
+
+    public static void createR1DistributionEquations(List<LfBranch> controllerBranches,
+                                                     EquationSystem<AcVariableType, AcEquationType> equationSystem) {
+        for (int i = 0; i < controllerBranches.size(); i++) {
+            LfBranch controllerBranch = controllerBranches.get(i);
+            // r1 at controller branch i
+            // r1_i = sum_j(r1_j) / controller_count where j are all the controller branches
+            // 0 = sum_j(r1_j) / controller_count - r1_i
+            // which can be rewritten in a more simple way
+            // 0 = (1 / controller_count - 1) * r1_i + sum_j(r1_j) / controller_count where j are all the controller branches except i
+            EquationTerm<AcVariableType, AcEquationType> r1 = equationSystem.getVariable(controllerBranch.getNum(), AcVariableType.BRANCH_RHO1)
+                    .createTerm();
+            Equation<AcVariableType, AcEquationType> zero = equationSystem.createEquation(controllerBranch.getNum(), AcEquationType.DISTR_RHO)
+                    .addTerm(r1.multiply(() -> 1d / controllerBranches.size() - 1));
+            for (LfBranch otherControllerBranch : controllerBranches) {
+                if (otherControllerBranch != controllerBranch) {
+                    EquationTerm<AcVariableType, AcEquationType> otherR1 = equationSystem.getVariable(otherControllerBranch.getNum(), AcVariableType.BRANCH_RHO1)
+                            .createTerm();
+                    zero.addTerm(otherR1.multiply(() -> 1d / controllerBranches.size()));
+                }
+            }
+        }
+    }
+
+    static void updateTransformerVoltageControlEquations(TransformerVoltageControl voltageControl, EquationSystem<AcVariableType, AcEquationType> equationSystem) {
+        boolean on = voltageControl.getMode() == DiscreteVoltageControl.Mode.VOLTAGE;
+
+        // activate voltage target equation if control is on
+        equationSystem.getEquation(voltageControl.getControlled().getNum(), AcEquationType.BUS_TARGET_V)
+                .orElseThrow()
+                .setActive(on);
+
+        List<LfBranch> controllerBranches = voltageControl.getControllers();
+        for (int i = 0; i < controllerBranches.size(); i++) {
+            LfBranch controllerBranch = controllerBranches.get(i);
+
+            // activate all rho1 equations if voltage control is off
+            equationSystem.getEquation(controllerBranch.getNum(), AcEquationType.BRANCH_TARGET_RHO1)
```

Comment:
Just to be sure that I understand well: here, later, I will have to take into account the local status of the branch controller, especially if the transformer is at its limits? But for now, we only use the status of the controlled bus.
In my previous tests, I have met a contingency that makes me lost one of the controller branches. This is supported by that PR ?